src/services/services.ts

```diff
@@ -1958,6 +1958,12 @@ namespace ts {
                     (char >= CharacterCodes.A && char <= CharacterCodes.Z) ||
                     (char >= CharacterCodes._0 && char <= CharacterCodes._9);
             }
+
+            function isNodeModulesFile(path: string): boolean {
+                const node_modulesFolderName = "/node_modules/";
+
+                return path.indexOf(node_modulesFolderName) !== -1;

```

Comment:
While I agree that the meanings of other negative values are unspecified, it seems borderline inconceivable that they could eventually be defined as success cases.  Ruling them out now seems more forward (and slightly-non-compliant implementation) compatible.